library/custom_template/custom_template.php

```diff
@@ -141,14 +141,115 @@ function refreshme() {
+            await includeScript(interactfn, false, 'script');
+        };
+        load().then(rtn => {
+            init();
         });
+    } else {
+        init();
+    }
+        function clearTranslate(elem){
+            elem.style.webkitTransform =
+                elem.style.transform =
+                'translate(' + 0 + 'px, ' + 0 + 'px)'
+            elem.setAttribute('data-x', 0)
+            elem.setAttribute('data-y', 0)
+        }
+        function switchElem(elem1, elem2, clear = false){
+            $(elem2).append($(elem1).children()[0]);
+            $(elem1).append($(elem2).children()[0]);
+            if(clear){
+                 clearTranslate($(elem2).children()[0]);
+                 clearTranslate($(elem1).children()[0]);
+            }
+        }
+        function moveUp(elem){
+            if(elem){
+                if($(elem).prev(".droppable").length > 0){
+                    let x = $(elem).prev(".droppable").children("li.is-dragging")[0];
+                    if(x){
+                        switchElem(elem, $(elem).prev(".droppable")[0], true);
+                        return true;
+                    }else{
+                        let c = $(elem).prev(".droppable")[0]; 
+                        if(c){
+                            let x = moveUp(c);
+                            if(x){
+                                switchElem(elem, $(elem).prev(".droppable")[0]);
+                            }
+                        }
+                    }
+                }
+            }
+            return false;
+        }
+        function moveDown(elem){
+            if(elem){
+                if($(elem).next(".droppable").length > 0){
+                    let x = $(elem).next(".droppable").children("li.is-dragging")[0];
+                    if(x){
+                        switchElem(elem, $(elem).next(".droppable")[0], true);
+                        return true;
+                    }else{
+                        let c = $(elem).next(".droppable")[0]; 
+                        if(c){
+                            let x = moveDown(c);
+                            if(x){
+                                switchElem(elem, $(elem).next(".droppable")[0]);
+                            }
+                        }
+                    }
+                }
+            }
+            return false;
+        }
+        function dragMoveListener (event) {
+            var target = event.target
+            var x = (parseFloat(target.getAttribute('data-x')) || 0) + event.dx
+            var y = (parseFloat(target.getAttribute('data-y')) || 0) + event.dy
+            target.style.webkitTransform =
+                target.style.transform =
+                'translate(' + x + 'px, ' + y + 'px)'
+            target.setAttribute('data-x', x)
+            target.setAttribute('data-y', y)
+        }
+        function init() {
+            interact('.droppable').dropzone({
```

Comment:
Lets see if we can add to utility.js as a callable for other use case in code and convenience.
maybe oeSortable(....)

Nice job on the JS

Reply:
I have moved it to utility and also made a few that i think would be nice but may not be desirable .Alos, i have added a callback parameter to the sortable function so we can link an event(if we want) required to take place whenver list changes.